torch/nn/utils/_named_members_polyfill.py

```diff
@@ -0,0 +1,32 @@
+# Polyfilled from pytorch core while we figure out the `remove_duplicate` issues.
+def _named_members(mod, get_members_fn, prefix='', recurse=True, remove_duplicate=True):
+    r"""Helper method for yielding various names + members of modules."""
+    memo = set()
+    modules = mod.named_modules(prefix=prefix) if recurse else [(prefix, mod)]
```

Comment:
This will fix the issue with the test but does mean that we won't warn on the case where we have a recursive module (i.e. `module = SomeModule(); module.m = module`) and a user passes in values for multiple of the recursive values